src/cpp/ext/otel/otel_plugin.h

```diff
@@ -384,6 +407,31 @@ class OpenTelemetryPlugin : public grpc_core::StatsPlugin {
     return plugin_options_;
   }
 
+  template <typename ValueType>
+  struct ObservableState {
+    void Observe(opentelemetry::metrics::ObserverResult& result)
+        ABSL_LOCKS_EXCLUDED(mu);
+
+    grpc_core::GlobalInstrumentsRegistry::InstrumentID id;
+    opentelemetry::nostd::shared_ptr<
+        opentelemetry::metrics::ObservableInstrument>
+        instrument;
+    bool callback_registered = false;
+    // We only support one Observable to one RegisteredMetricCallback and one
```

Comment:
Are you suggesting to store a map from `tuple<label_values, optional_label_values>` to metric value as the cache for each instrument and when the instrument's callback is fired, first do the cache update for any `RegisteredMetricCallback`s that are due and then feed the entire cache to OTel (i.e. call OTel's `Observe()` multiple times)? I think this might work.

Reply:
Uffff, I was thinking another way to store it is to have a map from `RegisteredMetricCallback*` to `tuple<value, label_values, optional_label_values>`, this way when a `RegisteredMetricCallback*` is removed, we can easily remove its corresponding entry from the cache.

Reply:
I just made a change in that direction, I haven't added any new tests yet. PTAL!

Reply:
I had actually been thinking of the former, but I like your approach better, because it does make it easier to remove old values when a callback is removed.